Manager/JobLauncher.php

```diff
@@ -0,0 +1,149 @@
+use Symfony\Component\EventDispatcher\GenericEvent;
+use Symfony\Component\Process\PhpExecutableFinder;
+use Symfony\Component\Security\Core\User\UserInterface;
+
+/**
+ * @author    Marie Bochu <marie.bochu@akeneo.com>
+ * @copyright 2015 Akeneo SAS (http://www.akeneo.com)
+ * @license   http://opensource.org/licenses/MIT MIT
+ */
+class JobLauncher
+{
+    /** @var ObjectManager */
+    protected $objectManager;
+
+    /** @var EventDispatcherInterface */
+    protected $eventDispatcher;
+
+    /** @var string */
+    protected $jobExecutionClass;
+
+    /** @var string */
+    protected $rootDir;
+
+    /** @var string */
+    protected $environment;
+
+    /** @var array */
+    protected $config = [];
+
+    /**
+     * Constructor
+     *
+     * @param ObjectManager            $objectManager
+     * @param EventDispatcherInterface $eventDispatcher
+     * @param string                   $jobExecutionClass
+     * @param string                   $rootDir
+     * @param string                   $environment
+     */
+    public function __construct(
+        ObjectManager $objectManager,
+        EventDispatcherInterface $eventDispatcher,
+        $jobExecutionClass,
+        $rootDir,
+        $environment
+    ) {
+        $this->objectManager      = $objectManager;
+        $this->eventDispatcher    = $eventDispatcher;
+        $this->jobExecutionClass  = $jobExecutionClass;
+        $this->rootDir            = $rootDir;
+        $this->environment        = $environment;
+    }
+
+    /**
+     * Launch a job with command
+     *
+     * @param JobInstance   $jobInstance
+     * @param UserInterface $user
+     * @param string        $rawConfiguration
+     *
+     * @return JobExecution
+     */
+    public function launch(JobInstance $jobInstance, UserInterface $user, $rawConfiguration = null)
+    {
+        $jobExecution = $this->create($jobInstance, $user);
+        $executionId  = $jobExecution->getId();
+        $pathFinder   = new PhpExecutableFinder();
+
+        if ($this->hasConfig('upload')) {
+            $rawConfiguration.= ' ' . addslashes(json_encode($jobInstance->getJob()->getConfiguration()));
+        }
+
+        $cmd = sprintf(
+            '%s %s/console akeneo:batch:job --env=%s %s %s %s %s >> %s/logs/batch_execute.log 2>&1',
+            $pathFinder->find(),
+            $this->rootDir,
+            $this->environment,
+            $this->hasConfig('email') ? sprintf('--email="%s"', $user->getEmail()) : '',
+            $jobInstance->getCode(),
+            $executionId,
+            !empty($rawConfiguration) ? sprintf('--config="%s"', $rawConfiguration) : '',
+            $this->rootDir
+        );
+
+        // Please note we do not use Symfony Process as it has some problem
+        // when executed from HTTP request that stop fast (race condition that makes
+        // the process cloning fail when the parent process, i.e. HTTP request, stops
+        // at the same time)
+        exec($cmd . ' &');
```

Comment:
Would be better to handle the command to another method, like launchInBackground() in order to allow easy override of the call (the '&' works only in Unix env) and better placement of comment.